llvm/test/Transforms/LoopVectorize/X86/limit-vf-by-tripcount.ll

```diff
@@ -258,26 +279,49 @@ exit:
 define void @limit_main_loop_vf_to_avoid_dead_main_vector_loop(ptr noalias %src, ptr noalias %dst) {
 ; CHECK-LABEL: @limit_main_loop_vf_to_avoid_dead_main_vector_loop(
 ; CHECK-NEXT:  entry:
-; CHECK-NEXT:    br i1 false, label [[SCALAR_PH:%.*]], label [[VECTOR_PH:%.*]]
+; CHECK-NEXT:    br i1 true, label [[SCALAR_PH:%.*]], label [[VECTOR_PH:%.*]]
```

Comment:
Doesn't this go against the wishes of the original author? It looks like we explicitly wanted to avoid having a dead main vector loop here - in fact this does seem like a regression. It's odd because the trip count seems to be 32 so with a VF=8 and IC=4 we should still be able to do one vector iteration.